internal/dag/gatewayapi_processor.go

```diff
@@ -195,39 +196,111 @@ func isSecretRef(certificateRef *gatewayapi_v1alpha1.LocalObjectReference) bool
 	return strings.ToLower(certificateRef.Kind) == "secret" && strings.ToLower(certificateRef.Group) == "core"
 }
 
-func (p *GatewayAPIProcessor) computeHosts(route *gatewayapi_v1alpha1.HTTPRoute) ([]string, []error) {
-	// Determine the hosts on the route, if no hosts
-	// are defined, then set to "*".
+func (p *GatewayAPIProcessor) computeHosts(route *gatewayapi_v1alpha1.HTTPRoute, listenerHostname *gatewayapi_v1alpha1.Hostname) ([]string, []error) {
```

Comment:
A comment explaining the logic of this method would be helpful, including the details around how the route and listener hostnames relate.

Reply:
👍